src/main/java/com/conveyal/r5/util/ExpandingMMFBytez.java

```diff
@@ -0,0 +1,233 @@
+package com.conveyal.r5.util;
+
+import com.conveyal.r5.transit.TransportNetwork;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.nustaq.offheap.bytez.BasicBytez;
+import org.nustaq.offheap.bytez.malloc.MallocBytez;
+import org.nustaq.serialization.*;
+import org.nustaq.serialization.coders.FSTBytezDecoder;
+import org.nustaq.serialization.coders.FSTBytezEncoder;
+import sun.nio.ch.FileChannelImpl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.lang.reflect.Method;
+import java.nio.channels.FileChannel;
+
+/**
+ * A byte buffer in a memory mapped file for use by the FST serialization library.
+ * https://github.com/RuedigerMoeller/fast-serialization
+ *
+ * Though Java allows memory-mapping a FileChannel to a ByteBuffer and FST has an (apparently experimental)
+ * ByteBufferBasicBytez implementation, ByteBuffers are addressed with signed ints so can store only 2GB.
+ * This introduces several levels of abstraction and truncates FST's internal long memory addresses.
+ *
+ * Here, we use the native methods on NIO FileChannel objects to allow 63-bit (signed long) addressing.
+ * Based on FST's MMFBytez and info from http://nyeggen.com/post/2014-05-18-memory-mapping-%3E2gb-of-data-in-java/
+ * Note that in practice on OSX you can't map >=4GB at once. Mapping 4GB - 4k does work though.
+ * http://lists.apple.com/archives/darwin-kernel/2008/Mar/msg00049.html
+ *
+ * This Bytez implementation implements all the methods needed for FST to expand it automatically.
+ * The alternative is to map a huge file and truncate it when done writing.
+ */
+public class ExpandingMMFBytez extends MallocBytez {
+
+    private static final Method map0;
+    private static final Method unmap0;
+    private static final int MAP_READ_WRITE = 1;
+
+    static {
+        try {
+            map0 = FileChannelImpl.class.getDeclaredMethod("map0", int.class, long.class, long.class);
```

Comment:
It's not, but it's the only way to get a long-addressable buffer. The real solution is streaming serialization in FST, for which there is a ticket, but it's not implemented yet. Added comments to the code.